### PR TITLE
fix: cannot run if link `analysis.sh` to another dir

### PR DESCRIPTION
**Pull request by archzi**
The `analysis.sh` cannot run when I link it to another directory. I drew inspiration from the implementation of JMeter. I hope this code can assist many individuals.

### CODE REVIEW

File: bin/analysis.sh [hunk 1]
```diff
@@ -17,8 +17,20 @@
...
+while [ -h "$PRG" ]; do
+	ls=`ls -ld "$PRG"`
+	link=`expr "$ls" : '.*-> \(.*\)$'`
+	if expr "$link" : '/.*' > /dev/null; then
+		PRG="$link"
+	else
+		PRG=`dirname "$PRG"`/"$link"
+	fi
+done
+base_dir=$(dirname "$PRG")
```
Comment (D-D-H) [inline bin/analysis.sh:33]: How about https://github.com/gradle/gradle-project-templates/blob/main/gradlew#L68 ?

Comment (archzi) [reply bin/analysis.sh:33]: hi, @D-D-H  Thanks for your comment. I have to say that this approach is better than JMeter's version. I will update the pr later

File: bin/analysis.sh [hunk 2]
```diff
@@ -17,8 +17,21 @@
...
+  [ -h "$PRG" ]
+do
+  ls=$( ls -ld "$PRG")
+  link=${ls#*' -> '}
+  case $link in
+    /* )  PRG=$link;;
+    *  )  PRG=$base_dir$link;;
+  esac
+done
+base_dir=$(dirname "$PRG")
```
Comment (D-D-H) [inline bin/analysis.sh:34]: base_dir=$( cd "${base_dir:-./}" && pwd -P ) || exit

Comment (archzi) [reply bin/analysis.sh]: Thanks for your advice, i've fixed it. Please review it.

**Review by D-D-H — APPROVED** [review]

**Pull request merged**